source/solvers/navier_stokes_scratch_data.cc

```diff
@@ -357,40 +393,83 @@ NavierStokesScratchData<dim>::calculate_physical_properties()
+                                                this->thermal_expansion_1[q]);
+
+                  const auto material_interaction_id =
+                    properties_manager.get_material_interaction_id(
+                      material_interactions_type::fluid_fluid, 0, 1);
+                  const auto mobility_ch_model =
+                    properties_manager.get_mobility_ch(material_interaction_id);
+                  mobility_ch_model->vector_value(fields, mobility_ch);
+                }
+              break;
             }
 
-
-          for (unsigned p = 0; p < previous_phase_values.size(); ++p)
+          if (gather_vof && !gather_ch) // VOF by default with two fluids
             {
               // Blend the physical properties using the VOF field
               for (unsigned int q = 0; q < this->n_q_points; ++q)
                 {
-                  // Calculate previous density (right now assumes constant
-                  // density model per phase)
-                  previous_density[p][q] = calculate_point_property(
-                    filter->filter_phase(this->previous_phase_values[p][q]),
-                    this->density_0[q],
-                    this->density_1[q]);
+                  double filtered_phase_value = this->filtered_phase_values[q];
+
+                  density[q] = calculate_point_property(filtered_phase_value,
+                                                        this->density_0[q],
+                                                        this->density_1[q]);
+
+                  viscosity[q] = calculate_point_property(filtered_phase_value,
+                                                          this->viscosity_0[q],
+                                                          this->viscosity_1[q]);
+
+                  thermal_expansion[q] =
+                    calculate_point_property(filtered_phase_value,
+                                             this->thermal_expansion_0[q],
+                                             this->thermal_expansion_1[q]);
                 }
+
+
+              for (unsigned p = 0; p < previous_phase_values.size(); ++p)
+                {
+                  // Blend the physical properties using the VOF field
+                  for (unsigned int q = 0; q < this->n_q_points; ++q)
+                    {
+                      // Calculate previous density (right now assumes constant
+                      // density model per phase)
+                      previous_density[p][q] = calculate_point_property(
+                        filter->filter_phase(this->previous_phase_values[p][q]),
+                        this->density_0[q],
+                        this->density_1[q]);
+                    }
+                }
+              break;
+            }
+
+          if (gather_ch && gather_vof)
+            {
+              throw std::runtime_error(
+                "Cannot solve a multiphase problem using VOF and Cahn-Hilliard at the same time");
```

Comment:
Good thinking! :)
However, this error could be catched even before in ``simulation_parameter.h``